src/function/aggregate/distributive/mode.cpp

```diff
@@ -9,81 +9,39 @@
 
 #include "duckdb/common/unordered_map.hpp"
 
-namespace duckdb {
+#include <functional>
 
-template <class T>
-struct ModeState {
-	unordered_map<T, size_t> *frequency_map;
-};
+namespace std {
 
-struct ModeFunction {
-	template <class STATE>
-	static void Initialize(STATE *state) {
-		state->frequency_map = nullptr;
+template <>
+struct hash<duckdb::interval_t> {
+	inline size_t operator()(const duckdb::interval_t &val) const {
+		return hash<int32_t> {}(val.days) ^ hash<int32_t> {}(val.months) ^ hash<int64_t> {}(val.micros);
 	}
+};
 
-	template <class INPUT_TYPE, class STATE, class OP>
-	static void Operation(STATE *state, FunctionData *bind_data, INPUT_TYPE *input, ValidityMask &mask, idx_t idx) {
-		if (!state->frequency_map) {
-			state->frequency_map = new unordered_map<INPUT_TYPE, size_t>();
-		}
-		(*state->frequency_map)[input[idx]]++;
+template <>
+struct hash<duckdb::hugeint_t> {
+	inline size_t operator()(const duckdb::hugeint_t &val) const {
+		return hash<int64_t> {}(val.upper) ^ hash<int64_t> {}(val.lower);
 	}
+};
 
-	template <class STATE, class OP>
-	static void Combine(STATE &source, STATE *target) {
-		if (!source.frequency_map) {
-			return;
-		}
-		if (!target->frequency_map) {
-			target->frequency_map = source.frequency_map;
-			source.frequency_map = nullptr;
-			return;
-		}
-		for (auto &val : *source.frequency_map) {
-			(*target->frequency_map)[val.first] += val.second;
-		}
-	}
+} // namespace std
 
-	template <class T, class STATE>
-	static void Finalize(Vector &result, FunctionData *, STATE *state, T *target, ValidityMask &mask, idx_t idx) {
-		if (!state->frequency_map) {
-			mask.SetInvalid(idx);
-			return;
-		}
-		//! Initialize control variables to first variable of the frequency map
-		auto highest_frequency = state->frequency_map->begin();
-		auto iterator = state->frequency_map->begin();
-		iterator++;
-		while (iterator != state->frequency_map->end()) {
-			if (iterator->second > highest_frequency->second) {
-				highest_frequency = iterator;
-			}
-			iterator++;
-		}
-		target[idx] = highest_frequency->first;
-	}
-	template <class INPUT_TYPE, class STATE, class OP>
-	static void ConstantOperation(STATE *state, FunctionData *bind_data, INPUT_TYPE *input, ValidityMask &mask,
-	                              idx_t count) {
-		for (idx_t i = 0; i < count; i++) {
-			Operation<INPUT_TYPE, STATE, OP>(state, bind_data, input, mask, 0);
-		}
-	}
+namespace duckdb {
 
-	static bool IgnoreNull() {
-		return true;
-	}
+bool operator==(const interval_t &lhs, const interval_t &rhs) {
+	return lhs.days == rhs.days && lhs.months == rhs.months && lhs.micros == rhs.micros;
+}
```

Comment:
Interval::Equals?

Reply:
or operator overloading in the timestamps_t?

Reply:
Sure - that would be my preference too!